src/index.js

```diff
@@ -0,0 +1,288 @@
+      setTimeout(function () {
+        closeModal(confirmModal);
+      }, 2000);
+    })
+    .catch(console.error)
+    .finally(() => {
+      editSubmitButton.textContent = "Save";
+    });
+}
+
+function handleAddCardSubmit(evt) {
+  evt.preventDefault();
+  const inputValues = { name: cardNameInput.value, link: cardLinkInput.value };
+  cardSubmitButton.textContent = "Saving...";
+  api
+    .addNewCard(inputValues)
+    .then((card) => {
+      const cardElement = getCardElement(card);
+      cardsList.prepend(cardElement);
+      closeModal(cardModal);
+      cardForm.reset();
+      disableButton(evt.submitter, settings);
+      confirmTitle.textContent = "New Post";
+      confirmMessage.textContent = "You have successfully created a post";
+      openModal(confirmModal);
+      setTimeout(function () {
+        closeModal(confirmModal);
+      }, 2000);
+    })
+    .catch(console.error)
+    .finally(() => {
+      cardSubmitButton.textContent = "Save";
+    });
+}
+
+changeAvatarForm.addEventListener("submit", (event) => {
+  event.preventDefault();
+
+  const formData = new FormData(event.target);
+  avatarSubmitButton.textContent = "Saving...";
+  const data = Object.fromEntries(formData);
+  api
+    .editAvatar(data)
+    .then(() => {
+      const avatar = document.querySelector(".profile__avatar");
+      avatar.src = data.avatar;
+      closeModal(changeAvatarModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      avatarSubmitButton.textContent = "Save"; // Reset to default text
+    });
+});
+
+// Card element generator
+function getCardElement(data) {
+  const cardElement = cardTemplate.content
+    .querySelector(".card")
+    .cloneNode(true);
+  const cardNameEl = cardElement.querySelector(".card__title");
+  const cardImgEl = cardElement.querySelector(".card__image");
+  const cardLikeBtn = cardElement.querySelector(".card__like-btn");
+  const cardDeleteBtn = cardElement.querySelector(".card__delete-btn");
+
+  cardImgEl.src = data.link;
+  cardImgEl.alt = data.name;
+  cardNameEl.textContent = data.name;
+
+  if (data.isLiked) {
+    cardLikeBtn.classList.add("card__like-btn_liked");
+  }
+
+  cardLikeBtn.addEventListener("click", () => {
+    const toggleLike = cardLikeBtn.classList.toggle("card__like-btn_liked");
+    toggleLike
+      ? api.addLike(data._id)
+      : api.removeLike(data._id).catch(console.error);
+  });
+
+  cardDeleteBtn.addEventListener("click", () => {
+    openModal(deleteConfirmModal);
+
+    const confirmHandler = () => {
+      api
+        .removeCard(data._id)
+        .then(() => {
+          cardElement.remove();
+          closeModal(deleteConfirmModal);
+          showConfirm("Delete Post", "You have successfully deleted a post");
+        })
+        .catch(console.error)
+        .finally(() => {
+          confirmDeleteBtn.removeEventListener("click", confirmHandler);
+        });
+    };
+
+    confirmDeleteBtn.addEventListener("click", confirmHandler);
```

Comment:
* Now all your cards (which you've tried to delete, but changed your mind) are **deleted** all at once. Please, try it yourself: add 3 cards in a row, then try to delete them, but don't do it (cancel and close the confirmation popup), and then delete any other card - and all your previous cards are deleted at once (all 3 and the other one). It’s because of a **memory leak**. All listeners are gathered in one element and they are fired all together when you confirm deleting. You need either to **correctly remove** submit listeners or update the card data.